CONTRIBUTION.md

```diff
@@ -28,58 +28,126 @@ We know that as of 1.12 this is not the case for great parts of the mod, but we
+  * The other approach would be using a `Map<SomePropertyOfMyObject, Consumer<MyObject>>` and setup all different cases as lambdas where the parameter is the object being tested.
+* Prefix Interfaces with `I` - like `ITemplate`.
+* `@SupressWarnings("...")` should be avoided in General:
+  * If it needs to be applied, limit it to the smallest possible scope (and create a new variable for it if necessary).
+  * An Suppression should always have a comment stating why it is valid to suppress this warning here.
+  * In general only a `@SupressWarning("unchecked")` should ever be necessary - do not suppress deprecations!
+* Capitalize comments.
+* Use multiline comments where applicable, and single-line-comments only for single line comments.
+* Write Javadoc instead of comments, where possible:
+  * `@implNote` and `@implSpec` are always possible...
+  * Try to list all unchecked exceptions with `@throws`
+  * Leave uncommented parameters out (no `@param obj` without description)
+* Avoid giant Method-Bodies where possible, by splitting into smaller Methods
 * Use Guava's `Preconditions` class for Argument validation
 * Validate `@NonNull` values with `Objects.requireNonNull` if necessary
 * use getters and setters, even in the declaring class
 * Use protected fields in non-API superclasses, rather than private ones with getters and setters, if:
-      1. Those field are barely accessed in that superclass and are heavily accessed in its subclasses.
-      2. There is no additional need for getters and setters, such as:
-          1. External access to the field.
-          2. The need to run some additional code or logic when getting or setting the value of the field.
+  1. Those field are barely accessed in that superclass and are heavily accessed in its subclasses.
+  2. There is no additional need for getters and setters, such as:
+    1. External access to the field.
+    2. The need to run some additional code or logic when getting or setting the value of the field.
 * Avoid non-private variables in API-Code and provide getters/setters instead. Bear in mind that this does not apply to constants (`public static final Type xyz = ...`) though.
 * In general try to use the smallest visibility-scope possible, expcept if it's likely that others will want access to the given method.
-* enforce Immutability where possible
-    * use ImmutableTypes for immutable values (`ImmutableList`, `ImmutableSet`, `ImmutableMap`, etc.)
-    * avoid public properties
-    
+* Enforce immutability where possible
+  * Use immutablet types for immutable values (`ImmutableList`, `ImmutableSet`, `ImmutableMap`, etc.)
+  * Avoid public properties
+* Always use capital letters for format suffixes and literals
+  * Bad format suffix: `0.5f`, `1000l`
+  * Good format suffiz: `0.5F`, `1000L`
+  * Bad iteral: `0xfff`, `0xbeef`
+  * Good literal: `0xFFF`, `0xBEEF`.
+* No nested classes with 2 levels or deeper.
```

Comment:
No and why?

Reply:
Are we talking about deep nested classes is ok or this is common sense?

Reply:
Last time I checked we're all Java programmers, We're not going to fail to understand mutiple levels of nested classes, sure, it might be a bit messy but it's not going to hurt the literal readbility of the code.